workflow.py

```diff
@@ -214,54 +215,114 @@ async def eliminate_subtraction(
     :param proc: number of processors to use
     :param results: the results to send to the api when the workflow is complete
     :param run_subprocess: runs a subprocess with error handling
-    :param subtraction: the subtraction to align and eliminate reads against
+    :param subtractions: the subtraction to align and eliminate reads against
     :param subtracted_sam_path: path to the SAM file with subtraction-mapped reads removed
     :param work_path: path to the workflow working directory
     """
     to_subtraction_sam_path = work_path / "to_subtraction.sam"
 
-    # Map reads to the subtraction.
-    await run_subprocess(
-        [
-            "bowtie2",
-            "--local",
-            "--no-unal",
-            "--no-hd",
-            "--no-sq",
-            "-N",
-            "0",
-            "-p",
-            str(proc),
-            "-x",
-            shlex.quote(str(subtraction.bowtie2_index_path)),
-            "-U",
-            str(isolate_fastq_path),
-            "-S",
-            str(to_subtraction_sam_path),
-        ]
-    )
+    working_isolate_path = work_path / "working_isolate.sam"
+    working_fastq_path = work_path / "working_fastq.fq"
 
-    await run_subprocess(
-        [
-            "./eliminate_subtraction",
-            str(isolate_sam_path),
-            str(to_subtraction_sam_path),
-            str(subtracted_sam_path),
-        ]
-    )
+    # copy the original isolate sam file into a working isolate sam file
+    # as to not disrupt uses elsewhere
+    shutil.copyfile(isolate_sam_path, working_isolate_path)
```

Comment:
1. This must be done with `await asyncio.to_thread(shutil.copyfile(...))` to avoid blocking our main thread. This goes for any blocking IO including file system access, HTTP requests, database requests, etc.
 
   [Background on Python threads if unfamiliar](https://realpython.com/intro-to-python-threading/#what-is-a-thread).

2. Why not just use `working_isolate_path = isolate_sam_path` before the first iteration then reassign at the end of each iteration? I'm not clear on the benefit of copying the file.